iothub/device/src/PayloadCollection.cs

```diff
@@ -114,6 +112,8 @@ public bool Contains(string key)
         /// <returns>True if a value of type <c>T</c> with the specified key was found; otherwise, it returns false.</returns>
         public bool TryGetValue<T>(string key, out T value)
         {
+            value = default;
```

Comment:
Best practice is to set an out parameter right away, so you are sure it is set no matter the code path. Then we don't need DRY violations below in the method either.